docs/current/cli/979595-reference.md

```diff
@@ -0,0 +1,181 @@
+---
+slug: /cli/979595/reference
+---
+
+# Reference
+
+The Dagger CLI provides a command-line interface to Dagger.
+
+## Usage
+
+```shell
+dagger [options] [command]
+```
+
+## Options
+
+The options below can be used with all CLI commands.
+
+| Option | Description |
+|---|---|
+| `--debug` | Show Buildkitd debug logs |
+| `-h`, `--help` | Show help text |
+| `--workdir` | Define the host working directory (default `.`) |
+|---|---|
+
+## Commands
+
+### dagger completion
+
+Generate the autocompletion script for dagger for the specified shell. Available shells are `bash`, `fish`, `zsh` and `powershell`.
+
+:::note
+You will need to start a new shell for this setup to take effect.
+:::
+
+#### Usage
+
+```shell
+dagger completion [shell]
+```
+
+#### Example
+
+Load completions for every new session on Linux:
+
+```shell
+dagger completion bash > /etc/bash_completion.d/dagger
+```
+
+To load completions for every new session on macOS:
+
+```shell
+dagger completion bash > $(brew --prefix)/etc/bash_completion.d/dagger
+```
+
+### dagger exec
+
+Executes the specified command in a Dagger session. `DAGGER_SESSION_URL` will be convieniently injected automatically.
+
+#### Usage
+
+```shell
+dagger exec [command]
+```
+
+#### Example
+
+Make an HTTP request using `curl`:
+
+```shell
+dagger exec -- sh -c 'curl \
+  -H "content-type:application/json" \
+  -d "{\"query\":\"{container{id}}\"}" \
+  http://$DAGGER_SESSION_URL/query'
+```
+
+### dagger help
+
+#### Usage
+
+```shell
+dagger help [command]
+```
+
+Retrieve detailed help text for any command in the application.
+
+#### Example
+
+Obtain help for the `dagger query` command:
+
+```shell
+dagger help query
+```
+
+### dagger listen
+
+Starts the Dagger Engine server.
+
+#### Usage
+
+```shell
+dagger listen [--listen address]`
+```
+
+#### Options
+
+| Option | Description |
+|---|---|
+| `--listen` | Listen on specified network address (default `8080`) |
+|---|---|
+
+#### Example
+
+Start Dagger Engine:
+
+```shell
+dagger listen
+```
+
+Start Dagger Engine on port 1234:
+
+```shell
+dagger listen --listen 1234
+```
+
```

Comment:
Same heads up as above: we may cut `dagger listen`

Reply:
OK